pkgs/by-name/fi/figurine/package.nix

```
@@ -0,0 +1,35 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+}:
+
+let
+  version = "1.3.0";
+in
+buildGoModule rec {
+  pname = "figurine";
+  inherit version;
```

```suggestion
buildGoModule rec {
  pname = "figurine";
  version = "1.3.0";
```

Ref https://github.com/NixOS/nixpkgs/pull/289152#discussion_r1491881815

---

disagree with this one only because it is used in `ldflags` and `version` below.

---

Hm what are the upside of using `let in` in your opinion here?

---

I am afraid I don't quite understand what you mean by "only defined in one place".
Might just be me not having English as first language.

So the review suggestion in question is:
```nix
buildGoModule rec {
  pname = "figurine";
  version = "1.3.0";
```

instead of
```nix
let
  version = "1.3.0";
in
buildGoModule rec {
  pname = "figurine";
  inherit version;
```

1. They do the same. Replacing one variant with another is a No-OP.
1. Having `let in` coupled with `rec` and then using `inherit version` for `version` and only `version` is a weird indirection that simply isn't needed.
1. The `rec` keyword and language feature is used by `src.rev` further down below in `ldflags` anyway.
1. `rec` is way more common across packages that use `buildGoModule` in nixpkgs.
1. There is more or less an unwritten guideline in nixpkgs, where lines and values that change with each update (here: `version`, `src.hash` and `vendorHash`) should stay as close together as possible.

Just to be extra sure: Do you know what the `rec` language feature does?
https://nixos.org/manual/nix/stable/language/constructs#recursive-sets

---

never mind, i was being stupid :)

i have made the requested changes. thanks for being so kind and patient!